Makefile

```diff
@@ -24,37 +26,58 @@ sedseparator = ''
 sedignorecase =
 endif
 
-build_dir = 'tmp/_output'
+build_dir := build/_output
 configcli_dest := $(build_dir)/configcli.tgz
 
 .DEFAULT_GOAL := build
 
 build: configcli pkg/apis/kubedirector.bluedata.io/v1alpha1/zz_generated.deepcopy.go | $(build_dir)
-	@echo
 	@echo \* Creating KubeDirector deployment image and YAML...
 	@test -d vendor || dep ensure -v
-	@echo operator-sdk build ${image}
-	@operator-sdk build ${image} | grep -v "Create deploy/operator.yaml"
+	operator-sdk build ${image}
 	@docker image prune -f > /dev/null
-	@sed -i ${sedseparator} \
-        -e '/command:/ {' \
-        -e 'n; ' \
-        -e 's~.*~          - "/bin/sh"~; G; ' \
-        -e 's~$$~          args:~; G; ' \
-        -e 's~$$~          - "-c"~; G; ' \
-        -e 's~$$~          - "mkfifo /tmp/fifo; (/root/${project_name} \&> /tmp/fifo) \& while true; do cat /tmp/fifo; done"~;' \
-        -e '}' deploy/operator.yaml
-	@sed -i ${sedseparator} \
+	@sed ${sedseparator} \
+        -e '/command:/ { n; ' \
+        -e 's~.*~          - "/bin/sh"~; ' \
+        -e 'a\          args:' \
+        -e 'a\          - "-c"' \
+        -e 'a\          - "mkfifo /tmp/fifo; (/root/${project_name} \&> /tmp/fifo) \& while true; do cat /tmp/fifo; done"' \
+        -e '}' \
         -e '/env:/ {' \
-        -e 'G; ' \
-        -e 's~$$~            - name: MY_NAMESPACE~; G; ' \
-        -e 's~$$~              valueFrom:~; G; ' \
-        -e 's~$$~                fieldRef:~; G; ' \
-        -e 's~$$~                  fieldPath: metadata.namespace~;' \
-        -e '}' deploy/operator.yaml
+        -e 'a\            - name: MY_NAMESPACE' \
+        -e 'a\              valueFrom:' \
+        -e 'a\                fieldRef:' \
+        -e 'a\                  fieldPath: metadata.namespace' \
+        -e '}' \
+        -e 's~REPLACE_IMAGE~${image}~' \
+        deploy/operator.yaml >${local_deploy_yaml}
+	@echo done
+	@echo
 
-	@echo "      serviceAccountName: kubedirector" >> deploy/operator.yaml
-	@mv deploy/operator.yaml deploy/kubedirector/deployment-localbuilt.yaml
+debug: export GOFLAGS += -gcflags=all=-N -gcflags=-l
+debug: build
+debug:
+	@echo \* Applying debug settings to KubeDirector deployment YAML...
+	@sed -i ${sedseparator} \
+        -e '/\/root\/${project_name}/s~~/root/dlv --listen=:40000 --headless --api-version=2 exec &~' \
+        -e '/template:/,/metadata:/ {' \
+        -e '/metadata:/ {' \
+        -e 'a\      annotations:' \
+        -e 'a\        container.apparmor.security.beta.kubernetes.io/${project_name}: unconfined' \
```

Comment:
for selinux enabled system, we may have to do selinux disabled as well

The equivalent docker run is this
--security-opt label=disable

But need to check on k8s

Reply:
These changes for relaxed security are only to allow a debugger to attach to kubedirector and I've successfully done that so I don't think this change is necessary. At least not at the moment.

Reply:
This reply is regarding Joel's comment about polling intervals above. I tried a quoted reply but it's at the bottom of this thread so this is a duplicate.

Good catch. That was on my todo list but I didn't get to it. I've checked in the changes but I have a couple of questions.

I currently have the polling cycle set to 30 secs for both the Config and Cluster CRs.

1. Is that polling cycle OK for the cluster CRs?

2. Can we not poll at all for the config CRs? There's no status to check and we can just reconcile when a change is posted.

Reply:
1. Yes that's what we had previously.

2. I think that's what the previous resyncPeriod of 0 meant for the config CRs.